packages/expo-intent-launcher/android/src/main/java/expo/modules/intentlauncher/IntentLauncherModule.kt

```diff
@@ -0,0 +1,134 @@
+package expo.modules.intentlauncher
+
+import expo.modules.core.Promise
+import expo.modules.core.ExportedModule
+import expo.modules.core.ModuleRegistry
+import expo.modules.core.interfaces.ExpoMethod
+import expo.modules.core.ModuleRegistryDelegate
+import expo.modules.core.arguments.ReadableArguments
+import expo.modules.core.interfaces.ActivityProvider
+import expo.modules.core.interfaces.services.UIManager
+import expo.modules.core.errors.ModuleNotFoundException
+import expo.modules.core.interfaces.ActivityEventListener
+import expo.modules.core.errors.CurrentActivityNotFoundException
+import expo.modules.intentlauncher.exceptions.ActivityAlreadyStartedException
+
+import android.net.Uri
+import android.os.Bundle
+import android.app.Activity
+import android.content.Intent
+import android.content.ComponentName
+import android.content.ActivityNotFoundException
+import android.content.Context
+
+private const val NAME = "ExpoIntentLauncher"
+
+private const val REQUEST_CODE = 12
+private const val ATTR_ACTION = "action"
+private const val ATTR_TYPE = "type"
+private const val ATTR_CATEGORY = "category"
+private const val ATTR_EXTRA = "extra"
+private const val ATTR_DATA = "data"
+private const val ATTR_FLAGS = "flags"
+private const val ATTR_PACKAGE_NAME = "packageName"
+private const val ATTR_CLASS_NAME = "className"
+
+class IntentLauncherModule(
+  context: Context,
+  private val moduleRegistryDelegate: ModuleRegistryDelegate = ModuleRegistryDelegate()
+) : ExportedModule(context), ActivityEventListener {
+  private var pendingPromise: Promise? = null
+  private val uiManager: UIManager? by moduleRegistry()
```

Comment:
```suggestion
  private val uiManager: UIManager by moduleRegistry()
```
We can assume that it isn't nullable